src/openms/source/IONMOBILITY/IMTypes.cpp

```diff
@@ -73,33 +73,45 @@ namespace OpenMS
     if (occs.empty())
     {
       return IMFormat::NONE;
-    }
-    if (occs.size() == 1 && (occs.find(IMFormat::CONCATENATED) != occs.end() || occs.find(IMFormat::MULTIPLE_SPECTRA) != occs.end()))
+    }    
+
+    if (occs.size() == 1) 
     {
-      return *occs.begin();
+      auto format = *occs.begin();
+      if(!(format != IMFormat::CONCATENATED 
+        || format != IMFormat::MULTIPLE_SPECTRA
+        || format != IMFormat::CENTROIDED))
+        {
+          throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "subfunction returned invalid value(s)", "Number of different values: " + String(occs.size()));
+        }
```

Comment:
not the same thing as before. It would throw if 3 values were returned, for example.

Reply:
The more serious issue I see is that we don't distinguish for the differetn MS level. But isn't it now better  (assuming that e.g. different level might have different representation and thus more than two formats would be possible?)

Reply:
this
```
if(!(format != IMFormat::CONCATENATED 
        || format != IMFormat::MULTIPLE_SPECTRA
        || format != IMFormat::CENTROIDED))
```
will never be true, for any input.
This basically says: `! (a || b || c)` <> ` (!a & !b & !c)` which means
``` 
if (format == IMFormat::CONCATENATED 
       && format == IMFormat::MULTIPLE_SPECTRA
        && format == IMFormat::CENTROIDED)
```
You meant 
```
if (format != IMFormat::CONCATENATED 
       && format != IMFormat::MULTIPLE_SPECTRA
        && format != IMFormat::CENTROIDED)
``` 
???

Reply:
``` "Number of different values: " + String(occs.size()));``  this will also always be 1.

But yes, if MS levels have different format, just returning `mixed` is probably a good thing.

Reply:
Can you add a test ?